server/data/data-api/src/main/java/org/apache/james/dlp/api/DLPRulesStore.java

```diff
@@ -0,0 +1,41 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.dlp.api;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.james.core.Domain;
+
+public interface DLPRulesStore {
+
+    Map<DLPRuleId, DLPRule> retrieveRules(Domain domain);
```

Comment:
we don't care  that much about Id, it should be a field of DLPRule and you should have a ```List<DLPRule> list()``` method IMO

Reply:
I agree

Reply:
That means that the user have to set the id as part of the request?

That seems like a bad idea to me! I'd rather not let him generate the id...

Reply:
I think we can remove the id field for now, or let the user generate it.

Reply:
> That seems like a bad idea to me! I'd rather not let him generate the id...

Why ? It's often the best solution to ensure idempotent property.